### DISCUSSION TITLE
Maneira correta de disponibilizar uma versão alterada com melhorias do sistema

### DISCUSSION THREAD

**Discussion by apoionacional** [RamonSilva20/mapos#2347] [General]
Inicialmente, parabéns pelo projeto.

Minha dúvida: Qual a maneira correta no github de disponibilizar as alterações (algumas de lógica bem drásticas) do sistema de maneira que eu possa contribuir de maneira intermitente com o projeto, sem descumprir as regras do autor?

Dentre outras alterações, foram feitas atualizações do Bootstrap para versão 5.3.3, Jquery-ui para 1.13.2, Jquery para 3.7.1, substituição da biblioteca font-awesome pela do próprio bootstrap, inclusão da definição de previsão padrão de orçamento em dias úteis nas configurações, remoção de diversos campos sem lógica (como por exemplo existência e/ou obrigação de definição de data final na entrada de um equipamento para reparo, inclusão de campos como série, voltagem..., data de autorização vinculada a definição de status como autorizado... Alteração de todo o template e otimização máxima para dispositivos móveis, alteração em todas as funções autoComplete substituindo if ($query->num_rows() > 0) por um loop foreach, função lower diretamente no sql, limitação dos caracteres exibidos em razão social recebidos do bd...

A bola da vez agora em minhas horas vagas é:
1 - Inclusão do recurso da assinatura na tela do dispositivo móvel vinculado a uma foto de face identificando quem autorizou, alterando automaticamente o status para autorizado ao recebe as duas imagens.

2 - Tratar o equipamento reparado como ativo de logo prazo do cliente (ex: Opção de cadastrar os equipamentos do cliente como ativo administrável, mesmo sem uma manutenção em andamento) com função de tratativas para RMA e função "contratos" de manutenção preventiva/corretiva.

3 - Inclusão de recurso de gerenciamento de postagem de retorno com confirmação de status por e-mail usando a api do melhor envios (Correis Sedex/Pac, latamcargo, Azul cargo, gollog).

4 - Inclusão de recurso de gerenciamento de fabricação de equipamentos com baixa de peças de reposição do estoque e geração de etiquetas do rotulador industrial pulando 100% a dependência de uso do sistema de etiquetadora industrial P-Touch (esse é meu maior desafio e sem precedentes).

Se alguém puder ajudar ou até mesmo contribuir na unha em conjunto seguindo essa linha de raciocínio que citei acima será muito bem vindo.
Meu forte é front-end e mediano no back-end.
80% javaScript, 50% TypeScript e 70% php.

Aos que tem dificuldade de instalação do projeto original (local ou hospedado online), me disponibilizo a ajudar nas horas vagas sem custo, é só chamar. É mais uma maneira de contribuir com esse projeto fabuloso.

A seguir a tela de inclusão de OS da versão anterior que fiz, no atual já fiz algumas alterações para inclusão do recurso de fotos de entrada do equipamento com inclusão da marca d'água de data e hora que foi tirado a foto, na visualização da OS tem opção de baixar como paginas secundarias do pdf com botão preto e branco ou colorida pra economia de tinta.

Exatamente como é exibido na tela de um S20 ultra que é o celular dos técnicos de campo:
![print_tela1](https://github.com/RamonSilva20/mapos/assets/108497625/c2b54efd-18a0-4a2d-b38e-1cc5d2b90069)

**Comment by Pr3d4dor** [comment]
Basta fazer PRs com as melhorias e as mesmas serão analisadas.
